google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ClientSideStatementValueConverters.java

```diff
@@ -85,6 +88,43 @@ public Boolean convert(String value) {
     }
   }
 
+  /** Converter from string to {@link Boolean} */
+  static class PgBooleanConverter implements ClientSideStatementValueConverter<Boolean> {
+
+    public PgBooleanConverter(String allowedValues) {}
+
+    @Override
+    public Class<Boolean> getParameterClass() {
+      return Boolean.class;
+    }
+
+    @Override
+    public Boolean convert(String value) {
+      if (value == null) {
+        return null;
+      }
+      if (value.length() > 1
+          && ((value.startsWith("'") && value.endsWith("'"))
+              || (value.startsWith("\"") && value.endsWith("\"")))) {
+        value = value.substring(1, value.length() - 1);
+      }
+      if ("true".equalsIgnoreCase(value)
+          || "tru".equalsIgnoreCase(value)
+          || "tr".equalsIgnoreCase(value)
+          || "t".equalsIgnoreCase(value)) {
+        return Boolean.TRUE;
+      }
+      if ("false".equalsIgnoreCase(value)
+          || "fals".equalsIgnoreCase(value)
```

Comment:
The values like `"tru", "fal", "tr"` are not supported without single quotes (`'`) but according to this code they will be supported without single quotes (`'`)

Reply:
Strangely enough, they are. I've tried a multitude of different statements with a real PG instance. See below for the results:

```sql
knut-test-db=# set default_transaction_read_only to "on";
SET

knut-test-db=# set default_transaction_read_only to tr;
SET

knut-test-db=# set default_transaction_read_only to ' true ';
ERROR:  parameter "default_transaction_read_only" requires a Boolean value

knut-test-db=# set default_transaction_read_only to "true";
SET

knut-test-db=# set default_transaction_read_only to 'true';
SET

knut-test-db=# set default_transaction_read_only to ye;
SET

knut-test-db=# set default_transaction_read_only to 'ye';
SET
```

So TLDR:
* Both single and double quotes are allowed (which is strange, as double quotes would normally be used to quote identifiers)
* Unique prefixes like `tr` and `n` are allowed both with and without quotes.
* Spaces inside quotes are __not__ allowed, so `' true '` is not a valid value.

Reply:
That's strange because once I tested inserting boolean values in some table and the results were totally opposite :) Btw thanks for clarifying